src/control/server/init/setup_spdk.sh

```diff
@@ -7,11 +7,39 @@
 
 set +e
 
+thisscriptname="$(basename "$0")"
+thisscriptpath="$(dirname "$(readlink -f "$0")")"
+
+echo "start of script: $thisscriptpath/$thisscriptname"
+
+rootdir="$(readlink -f "$(dirname "$0")")"/../..
+
+sname="setup.sh"
+setupscriptpath="$rootdir/spdk/scripts/$sname"
+if [ ! -x "$setupscriptpath" ]; then
+	if [ -x "/usr/share/spdk/scripts/$sname" ]; then
+		setupscriptpath="/usr/share/spdk/scripts/$sname"
+	else
+		echo "Could not find the SPDK $sname script" >&2
+		exit 1
+	fi
+fi
+
+ccname="config_converter.py"
+confconvertscriptpath="$rootdir/spdk/scripts/$ccname"
+if [ ! -x "$confconvertscriptpath" ]; then
+	if [ -x "/usr/share/spdk/scripts/$ccname" ]; then
+		confconvertscriptpath="/usr/share/spdk/scripts/$ccname"
+	else
+		echo "Could not find the SPDK $ccname script" >&2
+		exit 1
+	fi
+fi
```

Comment:
I am a bit wary of taking this approach, as the long-term goal is to avoid shelling out to and being dependent on these external scripts. Are things really in such a constant state of change that we can't handle the logic from these two scripts ourselves?

Reply:
Handling all the JSON options we might require sounds like more work than I have bandwidth for, the 2 files are both just over 700 lines so it's not insurmountable but I do feel like the cadence of change is high in the scripts and the maintenance of our implementations would be significant.